libs/wingc/src/type_check.rs

```diff
@@ -2334,19 +2334,14 @@ impl<'a> TypeChecker<'a> {
 				// Add members from the structs parents
 				let extends_types = extends
 					.iter()
-					.filter_map(|parent| match env.lookup(&parent, Some(stmt.idx)) {
-						Ok(kind) => match &*kind {
-							SymbolKind::Type(_type) => Some(*_type),
-							_ => {
-								self.type_error(TypeError {
-									message: format!("Expected {} to be a type", parent),
-									span: parent.span.clone(),
-								});
-								None
-							}
-						},
-						Err(type_error) => {
-							self.type_error(type_error);
+					.filter_map(|ext| {
+						let t = self
+							.resolve_user_defined_type(ext, env, stmt.idx)
+							.unwrap_or_else(|e| self.type_error(e));
+						if t.as_struct().is_some() {
+							Some(t)
+						} else {
+							self.general_type_error(format!("Expected a struct, instead found type \"{}\"", t));
```

Comment:
Random thought on error conventions: Is the word "instead" needed?

```suggestion
							self.general_type_error(format!("Expected a struct, found type \"{}\"", t));
```

Reply:
Interesting, yeah I think to my ears it still makes sense without "instead" 👍